examples/adc.rs

```diff
@@ -0,0 +1,72 @@
+#![deny(unsafe_code)]
+#![no_main]
+#![no_std]
+
+#[macro_use]
+extern crate cortex_m_rt as rt;
+extern crate cortex_m;
+extern crate cortex_m_semihosting;
+extern crate embedded_hal;
+extern crate panic_semihosting;
+extern crate stm32f1xx_hal;
+
+use core::fmt::Write;
+use cortex_m_semihosting::hio;
+use stm32f1xx_hal::prelude::*;
+
+use rt::ExceptionFrame;
+use stm32f1xx_hal::adc;
+
+#[entry]
+fn main() -> ! {
+    // aquire peripherals
+    let p = stm32f1xx_hal::stm32::Peripherals::take().unwrap();
+    let mut flash = p.FLASH.constrain();
+    let mut rcc = p.RCC.constrain();
+
+    // configure ADC clocks
+    let clocks = rcc.cfgr.adcclk(2.mhz()).freeze(&mut flash.acr);
```

Comment:
I don't think setting the ADC clock in an example is a good idea unless it's required for the operation of the ADC.

Reply:
Why not ? Default value is the slowest possible ADC clock (pclk2 / 8), just to be on the safe side and to ensure that 14MHz threshold is not exceeded. But ADC clock is configurable, so its frequency may be tweaked to meet certain practical needs, e.g. speed vs power savings choice.

Reply:
Alright, sounds reasonable. Though it would be nice to have a comment that explains that in the exammple